src/main/scala/nodes/learning/LeastSquaresEstimator.scala

```diff
@@ -0,0 +1,87 @@
+package nodes.learning
+
+import breeze.linalg._
+import nodes.util.{Densify, Sparsify}
+import org.apache.spark.rdd.RDD
+import pipelines.Logging
+import workflow.{Pipeline, WeightedNode, LabelEstimator, OptimizableLabelEstimator}
+
+import scala.reflect._
+
+/**
+ * A least squares solver that is optimized to use a fast algorithm, based on characteristics of the
+ * workload and cost models.
+ *
+ * Currently selects between Dense LBFGS, Sparse LBFGS, Exact Distributed Solve, and Approximate Block Solve.
+ *
+ * The default weights were determined empirically via results run on a 16 r3.4xlarge node cluster.
```

Comment:
Ah I see. I didn't realize we fit a linear model from experiments to get the constants. I somehow assumed we did this by running some microbenchmarks that say used raw network / mem-bw etc. This is more like the Planck / Ernest. I'm fine with pulling this R script into `scripts/` for now. We can return to setup a more automated way to do this in the future